build/Common.prod.props

```diff
@@ -1,6 +1,11 @@
 <Project>
   <Import Project=".\Common.props" />
 
+  <PropertyGroup>
+    <IsTrimmable>true</IsTrimmable>
+    <EnableAotAnalyzer>true</EnableAotAnalyzer>
```

Comment:
See my comment on https://github.com/open-telemetry/opentelemetry-dotnet/issues/3429#issuecomment-1430540405. 

This isn't actually doing anything because Open Telemetry isn't targeting `net6.0`+. The analyzers only work when the `System.*` assemblies are annotated with the correct attributes. And those attributes are only added to `net6.0` and `net7.0` TFMs.

Reply:
I think it is doing lots of things as the builds are failing.

And here is the target https://github.com/open-telemetry/opentelemetry-dotnet/blob/d0829fff2f229b67c884526cb4943b99f6c75f29/src/OpenTelemetry/OpenTelemetry.csproj#L4.

There are other components which might not target net6.0+, and that's something we should look into (maybe in a separate PR).

Reply:
Sorry - I should have said it isn't going to catch everything. Out of 21 projects under `src`, only 6 of them target net6:

![image](https://user-images.githubusercontent.com/8291187/219079161-874fe812-8440-4245-a25a-7402b6152064.png)

So, for example, the [issue that is blocking Open Telemetry from working in NativeAOT](https://github.com/open-telemetry/opentelemetry-dotnet/issues/3429) won't be caught by this because it occurs in `OpenTelemetry.Api` which only targets netstandard2.0 and net fx.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/a878ca3eead7d1b84c3e4c379c81b665b553c41b/src/OpenTelemetry.Api/OpenTelemetry.Api.csproj#L4

Reply:
We could have a build flag (ie -CompileForAnalyzers) that conditionally adds the latest framework to projects.
This would let us run other analyzers. The Microsoft.CodeAnalysis.NetAnalyzers is included in NET5+.